src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -807,15 +796,50 @@ public void DeleteInvalidBlock(Block invalidBlock)
 
             try
             {
-                DeleteBlocks(invalidBlock.Hash!);
+                DeleteInvalidBranch(invalidBlock.Hash!);
             }
             finally
             {
                 ReleaseAcceptingNewBlocks();
             }
         }
 
-        private void DeleteBlocks(Hash256 deletePointer)
+        public void DeleteBlock(long currentNumber, Hash256 currentHash, Hash256 nextHash, BatchWrite batch, ChainLevelInfo? currentLevel = null, bool isOldBlock = false)
+        {
+            currentLevel ??= LoadLevel(currentNumber);
+
+            bool shouldRemoveLevel = false;
+            if (currentLevel is not null) // preparing update of the level (removal of the invalid branch block)
+            {
+                if (currentLevel.BlockInfos.Length == 1)
+                {
+                    shouldRemoveLevel = true;
+                }
+                else
+                {
+                    currentLevel.BlockInfos = currentLevel.BlockInfos.Where(bi => bi.BlockHash != currentHash).ToArray();
+                }
+            }
+
+            UpdateDeletePointer(nextHash);
+
+            if (shouldRemoveLevel)
+            {
+                // only need to update BestKnownNumber if we are deleting most recent block level
+                if (!isOldBlock) BestKnownNumber = Math.Min(BestKnownNumber, currentNumber - 1);
+                _chainLevelInfoRepository.Delete(currentNumber, batch);
+            }
+            else if (currentLevel is not null)
+            {
+                _chainLevelInfoRepository.PersistLevel(currentNumber, currentLevel, batch);
+            }
+
+            _blockStore.Delete(currentNumber, currentHash);
+            _headerStore.Delete(currentHash);
+            if (isOldBlock) _blockInfoDb.Delete(currentHash);
```

Comment:
`_chainLevelInfoRepository` uses `_blockInfoDb`. So use `_chainLevelInfoRepository.Delete` is needed.